### PR TITLE
Issue 39840: Not prompting user about file overwrite when uploading multiple files

### PR DESCRIPTION
**Pull request by XingY**
#### Rationale
Overwrite has been set to true to silently overwrite when there is >1 files uploaded. This was probably due to tech difficulty in getting Ext.Msg box, which is a singleton, to show multiple instances at once.  Since user might be uploading a very large number of files with many many duplicates, we don't want to prompt file by file anyways. 

#### Related Pull Requests
* 

#### Changes
* change made to NOT set overwrite to true by default when uploading multiple files
* prompt for one or multiple files overwrite in a single dialog

### CODE REVIEW

File: filecontent/webapp/File/panel/Upload.js [hunk 1]
```diff
@@ -423,6 +412,47 @@ Ext4.define('File.panel.Upload', {
...
+                    else if (!dragDropError && alreadyExistFiles.length > 0) {
+                        var msg = '';
+                        if (alreadyExistFiles.length === 1)
+                            msg = "There is already a file named " + alreadyExistFiles[0].name + ' in this location. Would you like to replace it?';
+                        else if (alreadyExistFiles.length <=10) {
+                            msg = "The following files already exist in this location. Would you like to replace them?<br><ul>";
+                            Ext4.each(alreadyExistFiles, function(f) {
+                                msg += "<li>" + f.name + "</li>"
+                            }, this);
+                            msg += "</url>";
```
Comment (labkey-nicka) [inline filecontent/webapp/File/panel/Upload.js]: What is `</url>`? Did you mean `</ul>`?

File: filecontent/webapp/File/panel/Upload.js [hunk 2]
```diff
@@ -423,6 +412,47 @@ Ext4.define('File.panel.Upload', {
...
+                        }
+                        else {
+                            msg = "The following files already exist in this location. Would you like to replace them?<br><ul>";
+                            var count = 0;
+                            Ext4.each(alreadyExistFiles, function(f) {
+                                if (count++ >= 9)
+                                    return;
+                                msg += "<li>" + f.name + "</li>"
+                            }, this);
+                            msg += "</url>";
```
Comment (labkey-nicka) [inline filecontent/webapp/File/panel/Upload.js]: Same about `</url>`.

File: filecontent/webapp/File/panel/Upload.js [hunk 3]
```diff
@@ -423,6 +412,47 @@ Ext4.define('File.panel.Upload', {
...
 
                     }
+                    else if (!dragDropError && alreadyExistFiles.length > 0) {
+                        var msg = '';
+                        if (alreadyExistFiles.length === 1)
+                            msg = "There is already a file named " + alreadyExistFiles[0].name + ' in this location. Would you like to replace it?';
+                        else if (alreadyExistFiles.length <=10) {
+                            msg = "The following files already exist in this location. Would you like to replace them?<br><ul>";
+                            Ext4.each(alreadyExistFiles, function(f) {
+                                msg += "<li>" + f.name + "</li>"
```
Comment (labkey-nicka) [inline filecontent/webapp/File/panel/Upload.js]: I'm forgetting my `Ext4` but should this be:

```js
msg += "<li>" + Ext4.htmlEncode(f.name) + "</li>";
```

Same question for other HTML string concatenation in this change.

File: filecontent/webapp/File/panel/Upload.js [hunk 4]
```diff
@@ -359,6 +345,9 @@ Ext4.define('File.panel.Upload', {
                             if (file.errorText === 'Drag-and-drop upload') {
                                 dragDropError = true;
                             }
+                            else if (file.errorText === 'Already exists') {
```
Comment (labkey-nicka) [inline filecontent/webapp/File/panel/Upload.js]: nit: Basing the error processing of the error message is fragile. Is there a specific HTTP status that the server responds with in this case? I realize this was already being done for the "Drag-and-drop upload" but just thought I'd mention it.

**Review by labkey-nicka — CHANGES_REQUESTED** [review]

File: filecontent/webapp/File/panel/Upload.js (hunk 1 above)
Comment (XingY) [reply filecontent/webapp/File/panel/Upload.js]: yep..

**Review by labkey-nicka — APPROVED** [review]

**Comment by labkey-tchad** [comment]
@XingY I'm sorry for sitting on this for so long. Can I merge `develop` into this branch or do we want to try to get it into 20.7?

**Comment by XingY** [comment]
> @XingY I'm sorry for sitting on this for so long. Can I merge `develop` into this branch or do we want to try to get it into 20.7?

merge develop is probably fine. Doesn't sound like the client needed it urgently.

**Pull request merged**